python-threatexchange/threatexchange/api_models.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+from datetime import datetime
+from dateutil.parser import parse
+from dataclasses import dataclass
+
+"""
+Models (dataclasses only) for interfacing with the threatexchange API.
+"""
+
+
+def _parse_datetime_from_iso_8601(datestr: str) -> datetime:
+    """
+    Parses strings representing date like 2019-05-20T16:44:47+0000 from the
+    graph api into datetime objects.
+    """
+    return parse(datestr)
+
+
+@dataclass
+class ThreatPrivacyGroup:
+    id: str
+    name: str
+    description: str
+    members_can_see: bool
+    members_can_use: bool
+    last_updated: datetime
+
+    @classmethod
+    def from_graph_api_dict(cls, d: dict) -> "ThreatPrivacyGroup":
+        return cls(
```

Comment:
Be super super careful here. This isn't actually getting you any protection than using the dict directly (dict is erasing you all your types). This function isn't too too different than cls(**d).

Part of the reason why I didn't go the pytx/class route when I started on pytx3 is the original classes I wrote were all just giving me class method access to the same untyped json under the hood, and I didn't know enough about open source to figure out how to do typechecking.